src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/ModelBindingHelper.cs

```diff
@@ -636,17 +636,16 @@ internal static string GetPropertyName(Expression expression)
         /// Clears <see cref="ModelStateDictionary"/> entries for <see cref="ModelMetadata"/>.
         /// </summary>
         /// <param name="modelMetadata">The <see cref="ModelMetadata"/>.</param>
+        /// <param name="modelstate">The <see cref="ModelStateDictionary"/>.</param>
         /// <param name="modelKey">The entry to clear. </param>
-        /// <param name="modelMetadataProvider">The <see cref="IModelMetadataProvider"/>.</param>
         public static void ClearValidationStateForModel(
```

Comment:
I wonder if we'd be better off doing something really simple here - like clearing the whole dictionary if you give us the empty prefix. All of the stuff we can try is imprecise unless you used a prefix.